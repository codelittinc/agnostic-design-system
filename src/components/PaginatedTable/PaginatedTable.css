@import '../../styles/theme/index.css';

.paginated-footer .paginated-table-action-btns {
  align-items: var(--paginated-table-align-items, center);
  display: var(--paginated-table-buttons-display, inline-flex);
  justify-content: var(--paginated-table-justify-content, flex-end);
  margin: 0;
  overflow: var(--paginated-table-overflow, hidden);
  padding: 0;
}

.paginated-table-action-btns .btn-table-paginated {
  font-family: var(--paginated-table-font-family, 'Inter');
  font-size: var(--paginated-table-font-size, 14px);
  font-weight: var(--paginated-table-font-weight, 300);
  padding: var(--paginated-table-padding, 10px);
}

.number-of-pages {
  text-align: var(--paginated-table-page-number-text-align, end);
}

.number-of-pages-text:after {
  content: '|';
  margin-left: 20px;
}

.number-of-pages-input {
  --input-border: 5px solid red;
  --input-padding: 5px 10px;

  max-width: var(--paginated-table-page-input-max-width, 35px);
  display: inline-table;
  margin-right: 10px;
  text-align: center;
}

.pagination-items,
.number-of-pages,
.paginated-table-action-btns {
  vertical-align: baseline;
}
