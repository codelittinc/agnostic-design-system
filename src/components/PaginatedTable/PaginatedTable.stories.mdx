# PaginatedTable Component

## Type: Composite

The PaginatedTable contains the following components: [Table](https://design-system.codelitt.dev/?path=/docs/components-table--default), [Pagination](https://design-system.codelitt.dev/?path=/docs/components-pagination--default) and [NumericInput](https://design-system.codelitt.dev/?path=/docs/components-numeric-input--default).

## PaginatedTable Props

|        Prop        |              Type               |                                              Description                                              |        Default        | Mandatory |
| :----------------: | :-----------------------------: | :---------------------------------------------------------------------------------------------------: | :-------------------: | :-------: |
|   buttonNextIcon   |         React.ReactNode         |                           Displays the icon for the next pagination button.                           | <span> &rarr; </span> |   false   |
|   buttonPrevIcon   |         React.ReactNode         |                         Displays the icon for the previous pagination button.                         | <span> &larr; </span> |   false   |
|    currentPage     |             number              |                           Specifies the current page visible for the table.                           |           -           |   true    |
|  handlePageChange  |            function             |          Specifies what should happens when a change event is triggered on the page number.           |           -           |   true    |
|     onRowClick     |            function             | Specifies what happens when a user clicks in a specific row. Receives the row index and the row data. |           -           |   false   |
|     headerList     | array of strings or react nodes |                 Specifies the list to be used to populate the PaginatedTable header.                  |           -           |   true    |
| itemsOnCurrentPage | array of strings or react nodes |                      Specifies what should be the items displayed on the table.                       |           -           |   true    |
|       limit        |             number              |                     Specifies the maximum number of items each page should have.                      |           -           |   true    |
| totalNumberOfItems |             number              |                    Defines the total number of items to be displayed on the table.                    |           -           |   true    |
| variablesClassName |             string              |                   Specifies the custom class to apply styles to the PaginatedTable.                   |           -           |   false   |

## Getting Started

```javascript
import { PaginatedTable } from '@codelittinc/agnostic-design-system';

const props = {
  currentPage: 1,
  handlePageChange: e => e,
  headerList: ['First Name', 'Last Name', 'Handle', 'User Type', 'Status'],
  itemsOnCurrentPage: [
    ['John', 'Oto', '@timo', 'type1', 'Active'],
    ['Mark', 'Howard', '@mdo', 'type2', 'Inactive'],
    ['Jacob', 'Martin', '@mdo', 'type1', 'Active'],
    ['Lary', 'King', '@timo', 'type1', 'Inactive'],
    ['Helen', 'Bonham', '@timo', 'type2', 'Active'],
    ['Felix', 'Dougherty', '@timo', 'type1', 'Active'],
    ['Joseph', 'Thomas', '@timo', 'type1', 'Inactive'],
    ['Richard', 'Cardenas', '@timo', 'type1', 'Active']
  ],
  totalNumberOfItems: 15
};

const PaginatedTable = props => {
  return <PaginatedTable {...props} />;
};

export default PaginatedTable;
```

## CSS Variables Available

Given the nature of a composite component, in addition to its own, the PaginatedTable utilizes CSS variables from the Table, Pagination and NumericInput components.
PaginatedTable variables are listed in the table below.

### Variables of the PaginatedTable component

|      Property      |    Attribute    | State |
| :----------------: | :-------------: | :---: |
|                    |   align-items   |       |
|                    |   font-family   |       |
|                    |    font-size    |       |
|                    |   font-weight   |       |
|                    | justify-content |       |
|                    |     padding     |       |
|                    |    overflow     |       |
|      button        |   font-family   |       |
|      button        |    font-size    |       |
|      button        |   font-weight   |       |
|      button        |     padding     |       |
|      buttons       |     display     |       |
|   clickable-row    |     cursor      |       |
|      footer        |    font-size    |       |
|   footer-wrapper   |     display     |       |
|   footer-wrapper   |  flex-direction |       |
|    page-input      |    max-width    |       |
|    page-number     |   text-align    |       |
|     separator      |     color       |       |
|     separator      |     margin      |       |
| total-number-items |     margin      |       |
| total-number-items |  white-space    |       |
