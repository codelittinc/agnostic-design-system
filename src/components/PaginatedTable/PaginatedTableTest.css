@import '../../styles/theme/index.css';

.custom-table {
  --button-background-color: red;
  --button-color: black;
  --button-width: 69px;
  --button-height: 32px;
  --paginated-table-footer-wrapper-flex-direction: column;
  --paginated-table-page-number-text-align: normal;
  --paginated-table-page-number-text: 12px;
}
