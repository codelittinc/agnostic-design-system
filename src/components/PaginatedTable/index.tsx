import React, { useState } from 'react';
import NumericInput from '@/components/NumericInput';
import Pagination from '@/components/Pagination';
import Table from '@/components/Table';

import styles from '@/components/PaginatedTable/PaginatedTable.css';
import { isEmpty } from 'lodash';
export interface Props {
  currentPage: number;
  handlePageChange: (value: number) => void;
  headerList: React.ReactNode[];
  id?: string;
  itemsOnCurrentPage: React.ReactNode[][];
  limit: number;
  totalNumberOfItems: number;
  variablesClassName?: string;
}

const PaginatedTable = (props: Props) => {
  const {
    currentPage,
    handlePageChange,
    headerList,
    id,
    itemsOnCurrentPage,
    limit,
    totalNumberOfItems,
    variablesClassName
  } = props;

  const [uniqueId] = useState(id || Math.floor(Math.random() * 1000 + 1));

  const numberOfPages = Math.ceil(totalNumberOfItems / limit);

  const handleChange = event => {
    const { value } = event.target;

    if (value > 0 || isEmpty(value)) {
      handlePageChange(value);
    }
  };

  const itemsRangeBegin = () => {
    if (itemsOnCurrentPage) {
      return (currentPage - 1) * limit + 1;
    }
    return 0;
  };

  const itemsRangeEnd = () => {
    if (itemsOnCurrentPage) {
      const begin = itemsRangeBegin();
      const total = begin + itemsOnCurrentPage.length - 1;
      return Math.min(total, totalNumberOfItems);
    }
    return 0;
  };

  const widthFirstFooterCell = headerList.length - 2;

  return (
    <Table
      headerList={headerList}
      tableContent={itemsOnCurrentPage}
      variablesClassName={variablesClassName}
      renderFooter={() => (
        <tfoot className={styles['paginated-footer']}>
          <tr>
            <td className={styles['pagination-items']}>
              {itemsRangeBegin() !== 0
                ? `${itemsRangeBegin()} - ${itemsRangeEnd()} of ${totalNumberOfItems} items`
                : `-`}
            </td>
            <td className={styles['number-of-pages']} colSpan={widthFirstFooterCell}>
              <NumericInput
                id={`paginated-table-input-${uniqueId}`}
                max={numberOfPages}
                onChange={handleChange}
                onStateChange={() => {}}
                onFocus={() => {}}
                scale={0}
                size='medium'
                value={currentPage.toString()}
                variablesClassName={styles['number-of-pages-input']}
                positive
              />
              <span>of {numberOfPages} pages</span>
            </td>
            <td className={styles['paginated-table-action-btns']}>
              <Pagination
                totalPages={numberOfPages}
                numberOfPagesToShow={0}
                currentPage={currentPage}
                onSelectPage={handlePageChange}
                buttonNextLabel='Next'
                buttonNextIcon={<span> &rarr; </span>}
                buttonNextIconPosition='right'
                buttonPrevLabel='Prev'
                buttonPrevIcon={<span> &larr; </span>}
                buttonPrevIconPosition='left'
                buttonVariablesClassName={styles['btn-table-paginated']}
                variablesClassName={styles['paginated-table-pagination']}
              />
            </td>
          </tr>
        </tfoot>
      )}
    />
  );
};

export default PaginatedTable;

PaginatedTable.defaultProps = {
  headerList: [],
  tableContent: [[]]
};
