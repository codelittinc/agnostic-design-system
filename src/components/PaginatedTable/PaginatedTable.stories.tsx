import React, { useState } from 'react';
import PaginatedTable from '@/components/PaginatedTable';
import { Meta } from '@storybook/react';
import mdx from './PaginatedTable.stories.mdx';
import classNames from 'classnames';
import styles from './PaginatedTableTest.css';

export default {
  title: 'Components/PaginatedTable',
  component: PaginatedTable,
  parameters: {
    docs: {
      page: mdx
    }
  }
} as Meta;

const Template = args => {
  const [page, setPage] = useState(1);
  const listOfItems = args.listOfItems[page - 1];
  return (
    <PaginatedTable
      {...args}
      currentPage={page}
      handlePageChange={setPage}
      itemsOnCurrentPage={listOfItems}
    />
  );
};

const listOfItems = [
  [
    ['John', 'Oto', '@timo', 'type1', 'Active'],
    ['Mark', 'Howard', '@mdo', 'type2', 'Inactive'],
    ['Jacob', 'Martin', '@mdo', 'type1', 'Active'],
    ['Lary', 'King', '@timo', 'type1', 'Inactive'],
    ['Helen', 'Bonham', '@timo', 'type2', 'Active']
  ],
  [
    ['Bary', 'King', '@timo', 'type1', 'Inactive'],
    ['elen', 'Bonham', '@timo', 'type2', 'Active'],
    ['Felix', 'Dougherty', '@timo', 'type1', 'Active'],
    ['Joseph', 'Thomas', '@timo', 'type1', 'Inactive'],
    ['Richard', 'Cardenas', '@timo', 'type1', 'Active']
  ],
  [
    ['Ronis', 'Cardenas', '@timo', 'type1', 'Active'],
    ['Karl', 'Roy', '@twitter', 'type2', 'Active'],
    ['Sean', 'Steppe', '@timo', 'type1', 'Inactive'],
    ['Ruth', 'Cardenas', '@timo', 'type1', 'Active'],
    ['Ivan', 'Cardenas', '@timo', 'type1', 'Active']
  ]
];

export const Default = Template.bind({});
Default.args = {
  headerList: ['First Name', 'Last Name', 'Handle', 'User Type', 'Status'],
  totalNumberOfItems: 15,
  listOfItems: listOfItems,
  limit: 5
};

const listOfItemsIrregular = [
  [
    ['John', 'Oto', '@timo', 'type1', 'Active'],
    ['Mark', 'Howard', '@mdo', 'type2', 'Inactive'],
    ['Jacob', 'Martin', '@mdo', 'type1', 'Active'],
    ['Lary', 'King', '@timo', 'type1', 'Inactive'],
    ['Helen', 'Bonham', '@timo', 'type2', 'Active'],
    ['Bary', 'King', '@timo', 'type1', 'Inactive'],
    ['Bary', 'King', '@timo', 'type1', 'Inactive']
  ],
  [
    ['Bary', 'King', '@timo', 'type1', 'Inactive'],
    ['elen', 'Bonham', '@timo', 'type2', 'Active'],
    ['Felix', 'Dougherty', '@timo', 'type1', 'Active']
  ]
];

export const IrregularList = Template.bind({});
IrregularList.args = {
  headerList: ['First Name', 'Last Name', 'Handle', 'User Type', 'Status'],
  totalNumberOfItems: 10,
  listOfItems: listOfItemsIrregular,
  limit: 7
};

export const CustomizedTable = Template.bind({});
CustomizedTable.args = {
  headerList: ['First Name', 'Last Name', 'Handle', 'User Type', 'Status'],
  totalNumberOfItems: 15,
  listOfItems: listOfItems,
  limit: 5,
  variablesClassName: classNames(styles['custom-button'])
};
