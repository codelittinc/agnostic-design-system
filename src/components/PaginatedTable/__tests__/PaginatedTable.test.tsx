import React, { useState } from 'react';
import PaginatedTable, { Props } from '@/components/PaginatedTable';
import { render, screen, fireEvent, within } from '@testing-library/react';
import '@testing-library/jest-dom';

describe('PaginatedTable', () => {
  it('renders correctly', () => {
    const { asFragment } = render(
      <PaginatedTable
        currentPage={1}
        handlePageChange={() => {}}
        headerList={['First Name', 'Last Name', 'Handle', 'User Type', 'Status']}
        id='paginated-table'
        itemsOnCurrentPage={[
          ['John', 'Oto', '@timo', 'type1', 'Active'],
          ['Mark', 'Howard', '@mdo', 'type2', 'Inactive'],
          ['Jacob', 'Martin', '@mdo', 'type1', 'Active'],
          ['Lary', 'King', '@timo', 'type1', 'Inactive'],
          ['Helen', 'Bonham', '@timo', 'type2', 'Active']
        ]}
        limit={5}
        totalNumberOfItems={15}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  describe('Paginated input', () => {
    it('accepts the number of the page to display', () => {
      renderPaginatedTable();
      const PaginatedTableList = screen.getByTestId('table-element');
      const paginationInput = PaginatedTableList.querySelector('input')!;
      expect(paginationInput).toHaveValue('1');
      fireEvent.change(paginationInput, { target: { value: '3' } });
      expect(paginationInput).toHaveValue('3');
    });

    it('does not accept numbers smaller than 1', () => {
      renderPaginatedTable();
      const PaginatedTableList = screen.getByTestId('table-element');
      const paginationInput = PaginatedTableList.querySelector('input')!;
      expect(paginationInput).toHaveValue('1');
    });

    it('does not accept numbers bigger than the total number of pages', () => {
      renderPaginatedTable();
      const PaginatedTableList = screen.getByTestId('table-element');
      const paginationInput = PaginatedTableList.querySelector('input')!;
      expect(paginationInput).toHaveValue('1');
      fireEvent.change(paginationInput, { target: { value: '5' } });
      expect(paginationInput).toHaveValue('1');
    });
  });

  describe('Number of items displayed', () => {
    it('changes if you change the page', () => {
      renderPaginatedTable({
        itemsOnCurrentPage: [
          ['Mark', 'Howard', '@mdo', 'type2', 'Inactive'],
          ['Jacob', 'Martin', '@mdo', 'type1', 'Active'],
          ['Lary', 'King', '@timo', 'type1', 'Inactive'],
          ['Helen', 'Bonham', '@timo', 'type2', 'Active'],
          ['Felix', 'Dougherty', '@timo', 'type1', 'Active'],
          ['Joseph', 'Thomas', '@timo', 'type1', 'Inactive'],
          ['Richard', 'Cardenas', '@timo', 'type1', 'Active'],
          ['Karl', 'Roy', '@twitter', 'type2', 'Active'],
          ['Sean', 'Steppe', '@timo', 'type1', 'Inactive'],
          ['Samuel', 'Marriott', '@twitter', 'type1', 'Inactive']
        ],
        limit: 10
      });
      const PaginatedTableItems = screen
        .getByTestId('table-element')
        .querySelector('.pagination-items')!;

      const paginationInput = screen.getByTestId('table-element').querySelector('input')!;

      expect(PaginatedTableItems).toHaveTextContent(/1 - 10 of 15 items/);
      fireEvent.change(paginationInput, { target: { value: 2 } });
      expect(PaginatedTableItems).toHaveTextContent(/11 - 15 of 15 items/);
    });

    it('shows the right number of items per page', () => {
      renderPaginatedTable({
        itemsOnCurrentPage: [
          ['Mark', 'Howard', '@mdo', 'type2', 'Inactive'],
          ['Jacob', 'Martin', '@mdo', 'type1', 'Active'],
          ['Lary', 'King', '@timo', 'type1', 'Inactive'],
          ['Helen', 'Bonham', '@timo', 'type2', 'Active'],
          ['Felix', 'Dougherty', '@timo', 'type1', 'Active'],
          ['Joseph', 'Thomas', '@timo', 'type1', 'Inactive'],
          ['Richard', 'Cardenas', '@timo', 'type1', 'Active']
        ],
        limit: 7,
        totalNumberOfItems: 10
      });
      const PaginatedTableItems = screen
        .getByTestId('table-element')
        .querySelector('.pagination-items')!;

      const paginationInput = screen.getByTestId('table-element').querySelector('input')!;

      expect(PaginatedTableItems).toHaveTextContent(/1 - 7 of 10 items/);
      fireEvent.change(paginationInput, { target: { value: 2 } });
      expect(PaginatedTableItems).toHaveTextContent(/8 - 10 of 10 items/);
    });
  });

  describe('Usage of the prev button', () => {
    it('is disabled if you are on the first page', () => {
      renderPaginatedTable();
      const table = screen.getByTestId('table-element');
      const prevButton = within(table).getByText(/Prev/i).closest('button') as HTMLButtonElement;
      expect(prevButton.disabled).toBeTruthy();
    });

    it('is enabled if previous pages exist', () => {
      renderPaginatedTable({ currentPage: 2 });
      const table = screen.getByTestId('table-element');
      const prevButton = within(table).getByText(/Prev/i).closest('button') as HTMLButtonElement;
      expect(prevButton.disabled).not.toBeTruthy();
    });

    it('calls handlePageChange with the correct page number', () => {
      const handlePageChange = jest.fn();
      renderPaginatedTable({ handlePageChange, currentPage: 2 });
      fireEvent.click(screen.getByText(/Prev/));
      expect(handlePageChange).toHaveBeenCalledWith(1);
    });
  });

  describe('Usage of the next button', () => {
    it('is disabled if you are on the last page', () => {
      renderPaginatedTable({ currentPage: 3 });
      const table = screen.getByTestId('table-element');
      const nextButton = within(table).getByText(/Next/i).closest('button') as HTMLButtonElement;
      expect(nextButton.disabled).toBeTruthy();
    });

    it('is enabled if next pages exist', () => {
      renderPaginatedTable({ currentPage: 2 });
      const table = screen.getByTestId('table-element');
      const nextButton = within(table).getByText(/Next/i).closest('button') as HTMLButtonElement;
      expect(nextButton.disabled).not.toBeTruthy();
    });

    it('calls handlePageChange with the correct page number', () => {
      const handlePageChange = jest.fn();
      renderPaginatedTable({ handlePageChange, currentPage: 2 });
      fireEvent.click(screen.getByText(/Next/));
      expect(handlePageChange).toHaveBeenCalledWith(3);
    });
  });

  describe('Sortable table', () => {
    const SortableTable = () => {
      const [sortState, setSortState] = useState({});

      return (
        <div>
          <p data-testid='field'>Sort Field: {'sortField' in sortState && sortState.sortField}</p>
          <p data-testid='order'>Sort Order: {'sortOrder' in sortState && sortState.sortOrder}</p>

          <PaginatedTable
            currentPage={1}
            handlePageChange={() => {}}
            headerList={['First Name', 'Last Name', 'Handle', 'User Type', 'Status']}
            id='paginated-table'
            itemsOnCurrentPage={[
              ['John', 'Oto', '@timo', 'type1', 'Active'],
              ['Mark', 'Howard', '@mdo', 'type2', 'Inactive'],
              ['Jacob', 'Martin', '@mdo', 'type1', 'Active'],
              ['Lary', 'King', '@timo', 'type1', 'Inactive'],
              ['Helen', 'Bonham', '@timo', 'type2', 'Active']
            ]}
            limit={5}
            totalNumberOfItems={15}
            setSortState={setSortState}
          />
        </div>
      );
    };

    it('Displays the correct start sorting order and field', () => {
      render(<SortableTable />);
      const sortField = screen.getByTestId('field');
      const sortOrder = screen.getByTestId('order');
      expect(sortField).toHaveTextContent('Sort Field: First Name');
      expect(sortOrder).toHaveTextContent('Sort Order: ASC');
    });

    it('Displays the correct sorting field after it is updated', () => {
      render(<SortableTable />);
      fireEvent.click(screen.getByText(/Last Name/));
      const sortField = screen.getByTestId('field');
      expect(sortField).toHaveTextContent('Sort Field: Last Name');
    });

    it('Displays the correct sorting order after it is updated', () => {
      render(<SortableTable />);
      const secondHeaderElement = screen.getByText(/Last Name/);
      fireEvent.click(secondHeaderElement);
      fireEvent.click(secondHeaderElement);
      const sortField = screen.getByTestId('field');
      const sortOrder = screen.getByTestId('order');
      expect(sortField).toHaveTextContent('Sort Field: Last Name');
      expect(sortOrder).toHaveTextContent('Sort Order: DESC');
    });
  });
});

function Wrapper(props) {
  const [page, setPage] = useState(1);

  return (
    <PaginatedTable
      currentPage={page}
      id='paginated-table'
      itemsOnCurrentPage={[
        ['John', 'Oto', '@timo', 'type1', 'Active'],
        ['Mark', 'Howard', '@mdo', 'type2', 'Inactive'],
        ['Jacob', 'Martin', '@mdo', 'type1', 'Active'],
        ['Lary', 'King', '@timo', 'type1', 'Inactive'],
        ['Helen', 'Bonham', '@timo', 'type2', 'Active']
      ]}
      handlePageChange={setPage}
      headerList={['First Name', 'Last Name', 'Handle', 'User Type', 'Status']}
      totalNumberOfItems={15}
      limit={5}
      {...props}
    />
  );
}

function renderPaginatedTable(props: Partial<Props> = {}) {
  return render(<Wrapper {...props} />);
}
