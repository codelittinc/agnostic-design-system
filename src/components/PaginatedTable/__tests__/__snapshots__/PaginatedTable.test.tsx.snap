// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaginatedTable renders correctly 1`] = `
<DocumentFragment>
  <table
    class="table-container"
    data-testid="table-element"
  >
    <thead
      data-testid="table-header"
    >
      <tr>
        <th>
          First Name
        </th>
        <th>
          Last Name
        </th>
        <th>
          Handle
        </th>
        <th>
          User Type
        </th>
        <th>
          Status
        </th>
      </tr>
    </thead>
    <tbody
      data-testid="table-body"
    >
      <tr
        class="non-clickable-row"
        tabindex="0"
      >
        <td>
           John
        </td>
        <td>
           Oto
        </td>
        <td>
           @timo
        </td>
        <td>
           type1
        </td>
        <td>
           Active
        </td>
      </tr>
      <tr
        class="non-clickable-row"
        tabindex="0"
      >
        <td>
           Mark
        </td>
        <td>
           Howard
        </td>
        <td>
           @mdo
        </td>
        <td>
           type2
        </td>
        <td>
           Inactive
        </td>
      </tr>
      <tr
        class="non-clickable-row"
        tabindex="0"
      >
        <td>
           Jacob
        </td>
        <td>
           Martin
        </td>
        <td>
           @mdo
        </td>
        <td>
           type1
        </td>
        <td>
           Active
        </td>
      </tr>
      <tr
        class="non-clickable-row"
        tabindex="0"
      >
        <td>
           Lary
        </td>
        <td>
           King
        </td>
        <td>
           @timo
        </td>
        <td>
           type1
        </td>
        <td>
           Inactive
        </td>
      </tr>
      <tr
        class="non-clickable-row"
        tabindex="0"
      >
        <td>
           Helen
        </td>
        <td>
           Bonham
        </td>
        <td>
           @timo
        </td>
        <td>
           type2
        </td>
        <td>
           Active
        </td>
      </tr>
    </tbody>
    <tfoot
      class="paginated-footer"
    >
      <tr>
        <td
          colspan="5"
        >
          <div
            class="paginated-footer-wrapper"
          >
            <div
              class="pagination-items"
            >
              1 - 5 of 15 items
            </div>
            <div
              class="paginated-table-pages"
            >
              <div
                class="number-of-pages"
              >
                <div
                  class="number-of-pages-input container"
                >
                  <div
                    class="input--field-container"
                  >
                    <div
                      class="input--container"
                    >
                      <input
                        class="input input--medium"
                        id="paginated-table-input-paginated-table"
                        value="1"
                      />
                    </div>
                  </div>
                </div>
                <span
                  class="number-of-pages-text"
                >
                  of 3 pages
                </span>
              </div>
              <div
                class="paginated-table-action-btns"
              >
                <ul
                  class="pagination-list paginated-table-pagination"
                  data-testid="pagination-element"
                >
                  <li>
                    <button
                      class="button button--neutral button button--medium btn-table-paginated"
                      disabled=""
                      type="button"
                    >
                      <div
                        class="append-icon-left"
                      >
                        <div
                          class="button--append-icon"
                        >
                          <svg
                            data-file-name="SvgArrowLeft"
                          />
                        </div>
                      </div>
                      <span
                        class="button-text"
                      >
                        Prev
                      </span>
                    </button>
                  </li>
                  <li>
                    <button
                      class="button button--neutral button button--medium btn-table-paginated"
                      type="button"
                    >
                      <span
                        class="button-text"
                      >
                        Next
                      </span>
                      <div
                        class="append-icon-right"
                      >
                        <div
                          class="button--append-icon"
                        >
                          <svg
                            data-file-name="SvgArrowRight"
                          />
                        </div>
                      </div>
                    </button>
                  </li>
                </ul>
              </div>
            </div>
          </div>
        </td>
      </tr>
    </tfoot>
  </table>
</DocumentFragment>
`;
